bubble_sort.rb

```diff
@@ -0,0 +1,33 @@
+def bubble_sort(arr)
+  loop do
+    i = 1
+    test = false
+    while i - 1 < arr.length - 1
+      if arr[i - 1] > arr[i]
+        arr[i - 1], arr[i] = arr[i], arr[i - 1]
+        test = true
+      end
+      i += 1
+    end
+    break unless test
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  loop do
+    i = 1
+    test = false
+    while i - 1 < arr.length - 1
+      result = yield(arr[i - 1], arr[i])
+      if result.positive?
+        arr[i - 1], arr[i] = arr[i], arr[i - 1]
+        test = true
+      end
+      i += 1
+    end
+    break unless test
+  end
+  arr
+end
+
```

Comment:
check the results of the test you provided in your readme
```
p print bubble_sort_by([4, 3, 78, 2, 0, 2]) { |first, second| first <=> second }
```
```
[0, 2, 2, 3, 4, 78]nil
```
```
 p print bubble_sort_by(["hi","hello","hey"]) { |first, second| first <=> second }
```
```
["hello", "hey", "hi"]nil
```
```
p print bubble_sort_by(["hi","hello","hey"]) { |first, second| first.length - second.length }
```
```
["hi", "hey", "hello"]nil
```
There is a nil class appended to the array, let rectify this bug

Reply:
Dear TSE,
there is nothing wrong with our algorithm. The usage of the print function instead of p function to display the results always adds nil after the result. I guess Ruby is functioning that way. So we updated the tests in the readme file and put p instead of print. Please feel free to check the code again.
We will be waiting for your review again.